base_custom_info/models/custom_info.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+# © 2015 Antiun Ingeniería S.L. - Sergio Teruel
+# © 2015 Antiun Ingeniería S.L. - Carlos Dauden
+# © 2015 Antiun Ingeniería S.L. - Jairo Llopis
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import api, fields, models
+
+
+class CustomInfoTemplate(models.Model):
+    """Defines custom properties expected for a given database object."""
+    _name = "custom.info.template"
+    _description = "Custom information template"
+    _sql_constraints = [
+        ("name_model",
+         "UNIQUE (name, model_id)",
+         "Another template with that name exists for that model."),
+    ]
+
+    name = fields.Char(translate=True)
+    model_id = fields.Many2one(comodel_name='ir.model', string='Model')
+    info_ids = fields.One2many(
+        comodel_name='custom.info.property',
+        inverse_name='template_id',
+        string='Properties')
+
+
+class CustomInfoProperty(models.Model):
+    """Name of the custom information property."""
+    _name = "custom.info.property"
+    _description = "Custom information property"
+    _sql_constraints = [
+        ("name_template",
+         "UNIQUE (name, template_id)",
+         "Another property with that name exists for that template."),
+    ]
+
+    name = fields.Char(translate=True)
+    template_id = fields.Many2one(
+        comodel_name='custom.info.template',
+        string='Template')
+    info_value_ids = fields.One2many(
```

Comment:
I think you are confusing the model. This one is about properties: Size, color, etc. Since the same property is often shared between different objects, properties have their own table here.

This One2many allows the administrator to see how many values are set in any model with this property, it is not intended for end users.

Actual values are stored in `custom.info.value`, which has a property_id field pointing to this model to choose the property, a model_id + res_id fields combination to know where it applies to, and a value field to store that property's value.